openwisp_monitoring/device/models.py

```diff
@@ -78,22 +80,35 @@ def data_user_friendly(self):
             data['interfaces'].remove(interface)
         return data
 
+    def _get_db_points(self):
+        q = (
+            "SELECT data FROM {0}.{1} WHERE pk = '{2}' "
+            "ORDER BY time DESC LIMIT 1".format(SHORT_RP, self.__key, self.pk)
+        )
+        return list(query(q).get_points())
+
     @property
     def data(self):
         """
         retrieves last data snapshot from influxdb
         """
         if self.__data:
             return self.__data
-        q = (
-            "SELECT data FROM {0}.{1} WHERE pk = '{2}' "
-            "ORDER BY time DESC LIMIT 1".format(SHORT_RP, self.__key, self.pk)
-        )
-        points = list(query(q).get_points())
+        points = self._get_db_points()
         if not points:
             return None
```

Comment:
why don't you set `measured_at` here?

I'd also rename this attribute to 'data_timestamp'  (is it a timestamp right?).

That way we will avoid two queries to get the same information.